cli/tools/lint/ast_buffer/ts_estree.rs

```diff
@@ -721,11 +725,43 @@ impl TsEsTreeBuilder {
     span: &Span,
     ident: NodeRef,
     init: Option<NodeRef>,
+    definite: bool,
   ) -> NodeRef {
     let id = self.ctx.append_node(AstNode::VariableDeclarator, span);
 
     self.ctx.write_ref(AstProp::Id, &id, ident);
     self.ctx.write_maybe_ref(AstProp::Init, &id, init);
+    self.ctx.write_bool(AstProp::Definite, definite);
+
+    self.ctx.commit_node(id)
+  }
+
+  #[allow(clippy::too_many_arguments)]
+  pub fn write_ts_decl_fn(
```

Comment:
This could be renamed to match `write_fn_decl`

```suggestion
  pub fn write_ts_fn_decl(
```

Reply:
yeah, I'm neutral on that. The underlying AST node is named `TSDeclareFunction` in TSEStree.